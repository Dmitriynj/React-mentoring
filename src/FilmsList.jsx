import React from 'react';
import Grid from '@material-ui/core/Grid';
import { reduce, uniqueId, isEmpty } from 'lodash';
import { FilmCard } from './FilmCard';

const mockedFilmsList = [
  {
    title: 'Some',
    description: 'Description',
  },
  {
    title: 'Some1',
    description: 'Description1',
  },
  {
    title: 'Some1',
    description: 'Description2',
  },
  {
    title: 'Some3',
    description: 'Description3',
  },
  {
    title: 'Some4',
    description: 'Description4',
  },
  {
    title: 'Some5',
    description: 'Description5',
  },
  {
    title: 'Some6',
    description: 'Description6',
  },
  {
    title: 'Some7',
    description: 'Description7',
  },
  {
    title: 'Some8',
    description: 'Description8',
  },
  {
    title: 'Some9',
    description: 'Description8',
  },
  {
    title: 'Some10',
    description: 'Description8',
  },
  {
    title: 'Some11',
    description: 'Description8',
  },
  {
    title: 'Some12',
    description: 'Description8',
  },
  {
    title: 'Some13',
    description: 'Description8',
  },
];

const getFilmsPortion = (filmPortion) => {
  return filmPortion.map(({ title, description }) => (
    <Grid item xs={3} key={`film-row-item-${uniqueId()}`}>
      <FilmCard title={title} description={description} />
    </Grid>
  ));
};

const filmElements = reduce(
  mockedFilmsList,
  ({ result, temp }, filmData, index) => {
    if (index % 4 === 0 && !isEmpty(temp)) {
      return {
        result: [
          ...result,
          <Grid container item xs={12} key={`films-row-${uniqueId()}`}>
            {getFilmsPortion(temp)}
          </Grid>,
        ],
        temp: [filmData],
      };
    }

    if (index === mockedFilmsList.length - 1) {
      return {
        result: [
          ...result,
          <Grid container item xs={12} key={`films-row-${uniqueId()}`}>
            {getFilmsPortion([...temp, filmData])}
          </Grid>,
        ],
        temp: [],
      };
    }

    return {
      result,
      temp: [...temp, filmData],
    };
  },
  { result: [], temp: [] }
);

const FilmsList = () => {
  return <Grid container>{filmElements.result}</Grid>;
};

export { FilmsList };
