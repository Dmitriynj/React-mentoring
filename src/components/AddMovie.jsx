import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Input, InputLabel, Button } from '@material-ui/core';

const useStyles = makeStyles(() => ({
  field: {
    padding: '5px 0 5px 0',
  },
  input: {
    background: '#2e2e2e',
    color: 'white',
    fontWeight: 400,
    padding: '0 5px 0 5px',
    width: '100%',
    '&::after': {
      borderBottomColor: '#f65261',
    },
  },
  label: {
    fontWeight: 400,
    color: '#f65261',
    fontSize: 14,
    marginBottom: 2,
  },
  footer: {
    marginLeft: 'auto',
  },
  confirmButton: {
    marginLeft: 10,
    background: '#f65261',
    color: 'white',
    borderRadius: 4,
    border: '2px solid #f65261',
    '&:hover': { background: '#f65261' },
  },
  resetButton: {
    borderRadius: 4,
    border: '2px solid #f65261',
    color: '#f65261',
  },
}));

const AddMovie = () => {
  const classes = useStyles();

  return (
    <>
      <section className={classes.field}>
        <InputLabel className={classes.label}>MOVIE ID</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>TITLE</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>RELEASE DATE</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>MOVIE URL</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>GENRE</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>OVERVIEW</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel color="secondary" className={classes.label}>
          RUNTIME
        </InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.footer}>
        <Button className={classes.resetButton}>Reset</Button>
        <Button className={classes.confirmButton}>Submit</Button>
      </section>
    </>
  );
};

export { AddMovie };
