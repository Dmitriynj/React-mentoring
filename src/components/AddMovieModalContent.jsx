import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, Input, InputLabel } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { createMovie } from '../store/thunks';
import { MODAL_BUTTONS_STYLES } from '../constants';

const useStyles = makeStyles(() => ({
  field: {
    padding: '5px 0 5px 0',
  },
  input: {
    background: '#2e2e2e',
    color: 'white',
    fontWeight: 400,
    padding: '0 5px 0 5px',
    width: '100%',
    '&::after': {
      borderBottomColor: '#f65261',
    },
  },
  label: {
    fontWeight: 400,
    color: '#f65261',
    fontSize: 14,
    marginBottom: 2,
  },
  footer: {
    marginLeft: 'auto',
  },

  ...MODAL_BUTTONS_STYLES,
}));

const AddNewMovieStateless = ({ sendCreateReq, onConfirm }) => {
  const classes = useStyles();
  const [creatingMovie, setCreatingMovie] = useState({
    title: '',
    release_date: '',
    overview: '',
    genres: [''],
    runtime: '',
    poster_path: '',
  });

  console.log('creating movie', creatingMovie);

  const onChangeInput = (event) => {
    const { name } = event.target;
    let { value } = event.target;
    if (name === 'runtime') {
      value = Number(value);
    }
    if (name === 'genres') {
      value = value.split(',');
    }
    setCreatingMovie({ ...creatingMovie, [name]: value });
  };

  const onSubmitChanges = () => {
    sendCreateReq(creatingMovie);
    onConfirm();
  };

  return (
    <>
      <section className={classes.field} key="title-field">
        <InputLabel className={classes.label}>TITLE</InputLabel>
        <Input
          name="title"
          type="text"
          onChange={onChangeInput}
          value={creatingMovie.title}
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="release-date-field">
        <InputLabel className={classes.label}>RELEASE DATE</InputLabel>
        <Input
          name="release_date"
          onChange={onChangeInput}
          value={creatingMovie.release_date}
          type="date"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="genres-field">
        <InputLabel className={classes.label}>GENRES</InputLabel>
        <Input
          name="genres"
          onChange={onChangeInput}
          value={creatingMovie.genres.join(',')}
          type="text"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="overview-field">
        <InputLabel className={classes.label}>OVERVIEW</InputLabel>
        <Input
          name="overview"
          onChange={onChangeInput}
          value={creatingMovie.overview}
          type="text"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="runtime-field">
        <InputLabel className={classes.label}>RUNTIME</InputLabel>
        <Input
          name="runtime"
          onChange={onChangeInput}
          value={creatingMovie.runtime}
          type="number"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="poster-path-field">
        <InputLabel className={classes.label}>POSTER PATH</InputLabel>
        <Input
          name="poster_path"
          onChange={onChangeInput}
          value={creatingMovie.poster_path}
          type="text"
          className={classes.input}
        />
      </section>

      <section className={classes.footer} key="footer">
        <Button onClick={onSubmitChanges} className={classes.confirmButton}>
          Create
        </Button>
      </section>
    </>
  );
};
AddNewMovieStateless.propTypes = {
  sendCreateReq: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  sendCreateReq: (data) => dispatch(createMovie(data)),
});

const AddMovieModalContent = connect(null, mapDispatchToProps)(AddNewMovieStateless);

export { AddMovieModalContent };
