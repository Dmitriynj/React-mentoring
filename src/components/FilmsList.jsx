import React from 'react';
import Grid from '@material-ui/core/Grid';
import { reduce, uniqueId, isEmpty } from 'lodash';
import { FilmCard } from './FilmCard';

const mockedFilmsList = [
  {
    title: 'Some',
    description: 'Description',
    imageUrl: 'sample.jpg',
  },
  {
    title: 'Some1',
    description: 'Description1',
    imageUrl: 'sample1.jpg',
  },
  {
    title: 'Some1',
    description: 'Description2',
    imageUrl: 'sample2.jpg',
  },
  {
    title: 'Some3',
    description: 'Description3',
    imageUrl: 'sample3.jpg',
  },
  {
    title: 'Some4',
    description: 'Description4',
    imageUrl: 'sample4.jpg',
  },
  {
    title: 'Some5',
    description: 'Description5',
    imageUrl: 'sample5.jpg',
  },
  {
    title: 'Some6',
    description: 'Description6',
    imageUrl: 'sample6.jpg',
  },
  {
    title: 'Some7',
    description: 'Description7',
    imageUrl: 'sample7.jpg',
  },
  {
    title: 'Some8',
    description: 'Description8',
    imageUrl: 'sample8.jpg',
  },
  {
    title: 'Some9',
    description: 'Description8',
    imageUrl: 'sample9.jpg',
  },
  {
    title: 'Some10',
    description: 'Description8',
    imageUrl: 'sample10.jpg',
  },
  {
    title: 'Some11',
    description: 'Description8',
    imageUrl: 'sample11.jpg',
  },
  {
    title: 'Some12',
    description: 'Description8',
    imageUrl: 'sample12.jpg',
  },
];

const getFilmsPortion = (filmPortion) => {
  return filmPortion.map(({ title, description, imageUrl }) => (
    <Grid item xs={3} key={`film-row-item-${uniqueId()}`}>
      <FilmCard title={title} description={description} imageUrl={imageUrl} />
    </Grid>
  ));
};

const filmElements = reduce(
  mockedFilmsList,
  ({ result, temp }, filmData, index) => {
    if (index % 4 === 0 && !isEmpty(temp)) {
      return {
        result: [
          ...result,
          <Grid container item xs={12} key={`films-row-${uniqueId()}`}>
            {getFilmsPortion(temp)}
          </Grid>,
        ],
        temp: [filmData],
      };
    }

    if (index === mockedFilmsList.length - 1) {
      return {
        result: [
          ...result,
          <Grid container item xs={12} key={`films-row-${uniqueId()}`}>
            {getFilmsPortion([...temp, filmData])}
          </Grid>,
        ],
        temp: [],
      };
    }

    return {
      result,
      temp: [...temp, filmData],
    };
  },
  { result: [], temp: [] }
);

const FilmsList = () => {
  return <Grid container>{filmElements.result}</Grid>;
};

export { FilmsList };
