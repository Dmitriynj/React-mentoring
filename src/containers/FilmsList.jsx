import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import { connect } from 'react-redux';
import { reduce, uniqueId, isEmpty } from 'lodash';
import { fetchMovies } from '../store/thunks';
import { getMovies, getQueryOptions } from '../store/selectors';
import { FilmCard } from '../components/FilmCard';

const FilmsListStateless = ({ movies, queryOptions, getMovies }) => {
  useEffect(() => {
    getMovies(queryOptions);
  }, [queryOptions]);

  console.log('render', movies);

  const getFilmsPortion = (filmPortion) => {
    return filmPortion.map((curMovie) => (
      <Grid item xs={3} key={`film-row-item-${uniqueId()}`}>
        <FilmCard movie={curMovie} />
      </Grid>
    ));
  };

  const filmElements = reduce(
    movies,
    ({ result, temp }, filmData, index) => {
      if (index % 4 === 0 && !isEmpty(temp)) {
        return {
          result: [
            ...result,
            <Grid container item xs={12} key={`films-row-${uniqueId()}`}>
              {getFilmsPortion(temp)}
            </Grid>,
          ],
          temp: [filmData],
        };
      }

      if (index === movies.length - 1) {
        return {
          result: [
            ...result,
            <Grid container item xs={12} key={`films-row-${uniqueId()}`}>
              {getFilmsPortion([...temp, filmData])}
            </Grid>,
          ],
          temp: [],
        };
      }

      return {
        result,
        temp: [...temp, filmData],
      };
    },
    { result: [], temp: [] }
  );

  return <Grid container>{filmElements.result}</Grid>;
};

FilmsListStateless.propTypes = {
  movies: PropTypes.array.isRequired,
  getMovies: PropTypes.func.isRequired,
  queryOptions: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  movies: getMovies(state),
  queryOptions: getQueryOptions(state),
});

const mapDispatchToProps = (dispatch) => ({
  getMovies: (data) => dispatch(fetchMovies(data)),
});

const FilmsList = connect(mapStateToProps, mapDispatchToProps)(FilmsListStateless);

export { FilmsList };
