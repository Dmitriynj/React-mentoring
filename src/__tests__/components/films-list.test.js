import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { render } from '../test-utils';
import { FilmsList } from '../../components/FilmsList';

const movies = [
  {
    id: 1,
    title: 'some title1',
    release_date: '2021-01-01',
    poster_path: 'some poster path1',
    genres: ['Documentary', 'Comedy'],
    runtime: 1,
    overview: 'some text1',
  },
  {
    id: 2,
    title: 'some title2',
    release_date: '2021-01-02',
    poster_path: 'some poster path2',
    genres: ['Documentary'],
    runtime: 2,
    overview: 'some text2',
  },
  {
    id: 3,
    title: 'some title3',
    release_date: '2021-01-03',
    poster_path: 'some poster path3',
    genres: ['Documentary'],
    runtime: 3,
    overview: 'some text3',
  },
  {
    id: 4,
    title: 'some title4',
    release_date: '2021-01-04',
    poster_path: 'some poster path4',
    genres: ['Documentary'],
    runtime: 4,
    overview: 'some text4',
  },
  {
    id: 5,
    title: 'some title5',
    release_date: '2021-01-05',
    poster_path: 'some poster path5',
    genres: ['Documentary'],
    runtime: 5,
    overview: 'some text5',
  },
  {
    id: 6,
    title: 'some title6',
    release_date: '2021-01-06',
    poster_path: 'some poster path6',
    genres: ['Documentary'],
    runtime: 6,
    overview: 'some text6',
  },
  {
    id: 7,
    title: 'some title7',
    release_date: '2021-01-07',
    poster_path: 'some poster path7',
    genres: ['Documentary'],
    runtime: 7,
    overview: 'some text7',
  },
  {
    id: 8,
    title: 'some title8',
    release_date: '2021-01-08',
    poster_path: 'some poster path8',
    genres: ['Documentary'],
    runtime: 8,
    overview: 'some text8',
  },
];

function renderComponent() {
  return render(
    <MemoryRouter initialEntries={['/movies']}>
      <Route exact path="/movies">
        <FilmsList movies={movies} />
      </Route>
    </MemoryRouter>,
    { initialState: { moviesData: { totalAmount: 4, data: movies, offset: 0, limit: 10 } } }
  );
}

describe('Films list', () => {
  it('renders correctly', () => {
    const { container } = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });
});
