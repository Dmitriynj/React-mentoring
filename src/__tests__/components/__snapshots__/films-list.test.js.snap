// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films list renders correctly 1`] = `
<div
  class="MuiGrid-root MuiGrid-container"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title makeStyles-cardHeaderTitle-2 MuiTypography-h5 MuiTypography-displayBlock"
            >
              some title1
            </span>
          </div>
          <div
            class="MuiCardHeader-action"
          >
            <button
              aria-haspopup="true"
              aria-label="settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardMedia-root makeStyles-media-6"
          title="Contemplative Reptile"
        />
        <div
          class="MuiCardContent-root"
        >
          <div
            class="makeStyles-description-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              some text1
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Genres: 
            Documentary,Comedy
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Release date: 
            2021-01-01
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Avg vote: 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Budget: 
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title makeStyles-cardHeaderTitle-2 MuiTypography-h5 MuiTypography-displayBlock"
            >
              some title2
            </span>
          </div>
          <div
            class="MuiCardHeader-action"
          >
            <button
              aria-haspopup="true"
              aria-label="settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardMedia-root makeStyles-media-6"
          title="Contemplative Reptile"
        />
        <div
          class="MuiCardContent-root"
        >
          <div
            class="makeStyles-description-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              some text2
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Genres: 
            Documentary
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Release date: 
            2021-01-02
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Avg vote: 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Budget: 
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title makeStyles-cardHeaderTitle-2 MuiTypography-h5 MuiTypography-displayBlock"
            >
              some title3
            </span>
          </div>
          <div
            class="MuiCardHeader-action"
          >
            <button
              aria-haspopup="true"
              aria-label="settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardMedia-root makeStyles-media-6"
          title="Contemplative Reptile"
        />
        <div
          class="MuiCardContent-root"
        >
          <div
            class="makeStyles-description-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              some text3
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Genres: 
            Documentary
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Release date: 
            2021-01-03
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Avg vote: 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Budget: 
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title makeStyles-cardHeaderTitle-2 MuiTypography-h5 MuiTypography-displayBlock"
            >
              some title4
            </span>
          </div>
          <div
            class="MuiCardHeader-action"
          >
            <button
              aria-haspopup="true"
              aria-label="settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardMedia-root makeStyles-media-6"
          title="Contemplative Reptile"
        />
        <div
          class="MuiCardContent-root"
        >
          <div
            class="makeStyles-description-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              some text4
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Genres: 
            Documentary
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Release date: 
            2021-01-04
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Avg vote: 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Budget: 
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title makeStyles-cardHeaderTitle-2 MuiTypography-h5 MuiTypography-displayBlock"
            >
              some title5
            </span>
          </div>
          <div
            class="MuiCardHeader-action"
          >
            <button
              aria-haspopup="true"
              aria-label="settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardMedia-root makeStyles-media-6"
          title="Contemplative Reptile"
        />
        <div
          class="MuiCardContent-root"
        >
          <div
            class="makeStyles-description-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              some text5
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Genres: 
            Documentary
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Release date: 
            2021-01-05
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Avg vote: 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Budget: 
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title makeStyles-cardHeaderTitle-2 MuiTypography-h5 MuiTypography-displayBlock"
            >
              some title6
            </span>
          </div>
          <div
            class="MuiCardHeader-action"
          >
            <button
              aria-haspopup="true"
              aria-label="settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardMedia-root makeStyles-media-6"
          title="Contemplative Reptile"
        />
        <div
          class="MuiCardContent-root"
        >
          <div
            class="makeStyles-description-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              some text6
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Genres: 
            Documentary
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Release date: 
            2021-01-06
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Avg vote: 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Budget: 
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title makeStyles-cardHeaderTitle-2 MuiTypography-h5 MuiTypography-displayBlock"
            >
              some title7
            </span>
          </div>
          <div
            class="MuiCardHeader-action"
          >
            <button
              aria-haspopup="true"
              aria-label="settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardMedia-root makeStyles-media-6"
          title="Contemplative Reptile"
        />
        <div
          class="MuiCardContent-root"
        >
          <div
            class="makeStyles-description-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              some text7
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Genres: 
            Documentary
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Release date: 
            2021-01-07
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Avg vote: 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Budget: 
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title makeStyles-cardHeaderTitle-2 MuiTypography-h5 MuiTypography-displayBlock"
            >
              some title8
            </span>
          </div>
          <div
            class="MuiCardHeader-action"
          >
            <button
              aria-haspopup="true"
              aria-label="settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiCardMedia-root makeStyles-media-6"
          title="Contemplative Reptile"
        />
        <div
          class="MuiCardContent-root"
        >
          <div
            class="makeStyles-description-9"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              some text8
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Genres: 
            Documentary
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Release date: 
            2021-01-08
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Avg vote: 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Budget: 
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
